test/Microsoft.AspNet.Mvc.Core.Test/ParameterBinding/ModelBindingHelperTest.cs

```diff
@@ -466,6 +466,167 @@ public void GetPropertyName_FieldExpression_Throws()
+
+            var valueProvider = new DictionaryBasedValueProvider<TestValueBinderMetadata>(values);
+
+            // Act
+            var result = await ModelBindingHelper.TryUpdateModelAsync(
+                                                    model,
+                                                    model.GetType(),
+                                                    "",
+                                                    Mock.Of<HttpContext>(),
+                                                    modelStateDictionary,
+                                                    new DataAnnotationsModelMetadataProvider(),
+                                                    GetCompositeBinder(binders),
+                                                    valueProvider,
+                                                    validator,
+                                                    includePredicate);
+
+            // Assert
+            Assert.True(result);
+            Assert.Equal("MyPropertyValue", model.MyProperty);
+            Assert.Equal("IncludedPropertyValue", model.IncludedProperty);
+            Assert.Equal("Old-ExcludedPropertyValue", model.ExcludedProperty);
+        }
+
+        [Fact]
+        public async Task TryUpdateModelNonGeneric_ReturnsFalse_IfBinderReturnsFalse()
+        {
+            // Arrange
+            var metadataProvider = new Mock<IModelMetadataProvider>();
+            metadataProvider.Setup(m => m.GetMetadataForType(null, It.IsAny<Type>()))
+                            .Returns(new ModelMetadata(metadataProvider.Object, null, null, typeof(MyModel), null))
+                            .Verifiable();
+
+            var binder = new Mock<IModelBinder>();
+            binder.Setup(b => b.BindModelAsync(It.IsAny<ModelBindingContext>()))
+                  .Returns(Task.FromResult(false));
+            var model = new MyModel();
+
+            // Act
+            var result = await ModelBindingHelper.TryUpdateModelAsync(
+                                                    model,
+                                                    model.GetType(),
+                                                    null,
+                                                    Mock.Of<HttpContext>(),
+                                                    new ModelStateDictionary(),
+                                                    metadataProvider.Object,
+                                                    GetCompositeBinder(binder.Object),
+                                                    Mock.Of<IValueProvider>(),
+                                                    Mock.Of<IModelValidatorProvider>());
+
+            // Assert
+            Assert.False(result);
+            Assert.Null(model.MyProperty);
+            metadataProvider.Verify();
+        }
+
+        [Fact]
+        public async Task TryUpdateModelNonGeneric_ReturnsTrue_IfModelBindsAndValidatesSuccessfully()
```

Comment:
Is the difference between this and ```TryUpdateModelNonGeneric_ModelTypeOverload_ReturnsTrue_ModelBindsAndValidatesSuccessfully``` that the later is using predicate overload? Can we capture it as test name?